File: retort-core/src/main/java/co/kaioru/retort/builder/IMiddlewareBuilder.java
package co.kaioru.retort.builder;

import co.kaioru.retort.ICommandContext;
import co.kaioru.retort.ICommandExecutable;
import co.kaioru.retort.ICommandMiddleware;

public interface IMiddlewareBuilder<I extends ICommandContext> {

    IMiddlewareBuilder<I> withName(String name);

    IMiddlewareBuilder<I> withDescription(String description);

    ICommandMiddleware<I> build(ICommandExecutable<I, Boolean> executable);

}

File: retort-core/src/main/kotlin/co/kaioru/retort/builder/IMiddlewareBuilder.kt
package co.kaioru.retort.builder

import co.kaioru.retort.ICommandExecutable
import co.kaioru.retort.ICommandMiddleware

interface IMiddlewareBuilder<I> : ICommandMiddleware<I> {
    fun build(executable: ICommandExecutable<I, Boolean>): ICommandMiddleware<I>
}